shared/constants/transaction.js

```diff
@@ -0,0 +1,95 @@
+export const TRANSACTION_STATUS_SUBMITTED = 'submitted'
+export const TRANSACTION_STATUS_FAILED = 'failed'
+export const TRANSACTION_STATUS_DROPPED = 'dropped'
+
+// Transaction Category is a MetaMask construct used internally
+// Used to categorize transactions by purpose for displaying information to the user
+// 1. `'tranfser'`            - A token transaction where the user is sending tokens that they
+//                              own to another address
+// 2. `'transferfrom'`        - A token transaction transferring tokens from an account that
+//                              the sender has an allowance of (see the approve category).
+// 3. `'approve'`             - A token transaction requesting an allowance of the token to
+//                              spend on behalf of the user
+// 4. `'incoming'`            - An incoming (deposit) transaction
+// 5. `'sentEther'`           - A transaction sending ether to a recipient
+// 6. `'contractInteraction'` - A transaction that is interacting with a smart contract's methods
+// 7. `'contractDeployment'`  - A transaction that deployed a smart contract
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER = 'transfer'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_APPROVE = 'approve'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER_FROM = 'transferfrom'
+export const TRANSACTION_CATEGORY_INCOMING = 'incoming'
+export const TRANSACTION_CATEGORY_SENT_ETHER = 'sentEther'
+export const TRANSACTION_CATEGORY_CONTRACT_INTERACTION = 'contractInteraction'
+export const TRANSACTION_CATEGORY_DEPLOY_CONTRACT = 'contractDeployment'
+
+// Transaction Group Status is a MetaMask construct to track status of groups
+// of transactions.
+// 1. `'cancelled'` - A cancel type transaction in the group was confirmed
+// 2. `'pending'`   - The primaryTransaction of the group has a status that is one of:
+//                    1. TRANSACTION_STATUS_APPROVED
+//                    2. TRANSACTION_STATUS_UNAPPROVED
+//                    3. TRANSACTION_STATUS_SUBMITTED
+export const TRANSACTION_GROUP_STATUS_CANCELLED = 'cancelled'
+export const TRANSACTION_GROUP_STATUS_PENDING = 'pending'
+
+// Transaction Group Category is a MetaMask construct to categorize the intent
+// of a group of transactions for purposes of displaying in the UI
+// 1. `'send'`              - Transaction group representing ether being sent from the user
+// 2. `'receive'`           - Transaction group representing a deposit/incoming transaction
+//                            usually maps 1:1 with TRANSACTION_CATEGORY_INCOMING
+// 3. `'interaction'`       - Transaction group representing an interaction with a smart
+//                            contract's methods.
+// 4. `'approval'`          - Transaction group representing a request for an allowance
+//                            of a token to spend on the user's behalf
+// 5. `'signature-request'` - Transaction group representing a signature request
```

Comment:
In an ideal world, these would not be included here. I don't think we should call anything a "transaction" that is not a transaction, in the code or the UI. It's unfortunate that this is how the signature requests were integrated. This is responsible for a lot of existing complicated UI logic.

It would seem more appropriate to have a higher-level abstraction (e.g. "activity"), something that could be either a transaction or a signature request/signing, or something else. The set of things we want to show in the activity tab.

Reply:
Agreed. Although at this point in the code the things that get returned from nonceSortedTransactionSelector are called `TransactionGroup`, I intend on changing that in a future PR and updating these to reflect that. Also ideally I'll at some point move this logic into a background controller, enabling things like *real* pagination of activity, and that would make these variables -- whatever they end up being called -- make sense in a shared constant file.